http/vertx-web-validation/src/main/java/io/quarkus/ts/vertx/web/validation/ShopResource.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.ts.vertx.web.validation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/shoppinglist")
+@Produces(MediaType.APPLICATION_JSON)
+public class ShopResource {
+
+    private List<ShoppingList> shoppingList;
+
+    private List<ShoppingList> createSampleProductList() {
+        shoppingList = new ArrayList<>();
+        shoppingList.add(new ShoppingList(UUID.randomUUID(), "ListName1", 25,
+                new ArrayList<>(Arrays.asList("Carrots", "Water", "Cheese", "Beer"))));
+        shoppingList.add(new ShoppingList(UUID.randomUUID(), "ListName2", 80,
+                new ArrayList<>(Arrays.asList("Meat", "Wine", "Almonds", "Potatoes", "Cake"))));
+        return shoppingList;
+    }
+
+    @GET
+    public List<ShoppingList> get() {
+        if (shoppingList == null) {
+            createSampleProductList();
```

Comment:
This can only work because of how you use it, it's not a good pattern to assign value to non-volatile variable inside exposed method that can be called concurrently. It won't cause us issues unless someone rewrites your tests or add a new ones... so it can be kept.

Just note that I don't like it :-)